### ISSUE TITLE
plugin isn't compatible with 2018.x EAP releases

### ISSUE THREAD

**Issue by vvuk** [cursive-ide/emacs-plus#3]
The plugin has `until-build="173.*"`, so compat stops with 2017.3.  I don't actually *know* whether it's really compatible with 2018, but I'll build and try it as soon as I can.

**Comment by crossroad0201** [comment]
An error occuered on IDEA 2018.1.

```
IntelliJ IDEA 2018.1 (Ultimate Edition)
Build #IU-181.4203.550, built on March 27, 2018
JRE: 1.8.0_152-release-1136-b20 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.13.3
```

```
Argument for @NotNull parameter 'name' of com/mulgasoft/emacsplus/EmacsPlus.resetCommand must not be null
java.lang.IllegalArgumentException: Argument for @NotNull parameter 'name' of com/mulgasoft/emacsplus/EmacsPlus.resetCommand must not be null
	at com.mulgasoft.emacsplus.EmacsPlus.resetCommand(EmacsPlus.java)
	at cursive.emacsplus.EmacsPlus$1.commandFinished(EmacsPlus.java:37)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.fireCommandFinished(CoreCommandProcessor.java:214)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.finishCommand(CoreCommandProcessor.java:187)
	at com.intellij.openapi.command.impl.CommandProcessorImpl.finishCommand(CommandProcessorImpl.java:53)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:155)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:109)
	at com.intellij.diff.util.DiffUtil.lambda$executeWriteCommand$12(DiffUtil.java:1337)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1010)
	at com.intellij.diff.util.DiffUtil.executeWriteCommand(DiffUtil.java:1336)
	at com.intellij.diff.util.DiffUtil.executeWriteCommand(DiffUtil.java:1354)
	at com.intellij.openapi.vcs.ex.LineStatusTrackerBase.updateDocument(LineStatusTrackerBase.kt:140)
	at com.intellij.openapi.vcs.ex.LineStatusTrackerBase.updateDocument(LineStatusTrackerBase.kt:131)
	at com.intellij.openapi.vcs.ex.LineStatusTrackerBase.setBaseRevision(LineStatusTrackerBase.kt:107)
	at com.intellij.openapi.vcs.ex.PartialLocalLineStatusTracker.setBaseRevision(PartialLocalLineStatusTracker.kt:154)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager$MyBaseRevisionLoader.handleResult(LineStatusTrackerManager.kt:582)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager$MyBaseRevisionLoader.handleResult(LineStatusTrackerManager.kt:479)
	at com.intellij.openapi.vcs.impl.SingleThreadLoader$handleSingleRequest$$inlined$runInEdt$1.run(actions.kt:70)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:447)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:431)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:415)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:837)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:766)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:434)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by nessus42** [comment]
I hope that this is fixed to work with IntelliJ 2018.x sometime soon. I can't live without Emacs+!

**Comment by MrFroop** [comment]
I tested to rebuild the plugin with the newest IntellijCE and changed the supported version to "181.*". And that seems to be all that needs to be done for it to work again.

**Comment by cursive-ide** [comment]
I just published 3.700 which is compatible with 2018.1 and also fixes the NPE above.

**Comment by ningliu0** [comment]
@cursive-ide 3.700 is incompatible with the new release 2018.2. Can you please build a new version? Thanks.

**Comment by cursive-ide** [comment]
@ningliu0 I've just released 0.3.800 which supports 2018.2.
